docs/react/README.md

```diff
@@ -1,72 +1,186 @@
+`hatchifyReact(createJsonapiClient("/api", Schemas))` is a `Function` that initializes the `HatchifyApp` object from the JSON:API rest client. Inside of the retured object you will find [`components`](./components.md), [`model`](./model.md), [`state`](./state.md), and [`Everything`](#everything).
 
 ```ts
+import {createJsonapiClient, hatchifyReact} from "@hatchifyjs/react"
+
+const schemas = { ... }
+
 const hatchedReact = hatchifyReact(createJsonapiClient("/api", Schemas))
 ```
 
-## HatchifyProvider
+### HatchifyProvider
 
-`HatchifyProvider` is a component that hosts and provides access to Hatchify-related state. Hatchify components in `HatchifyProvider`'s subtree interact with it to access Hatchify internal state. It must be a parent to any Hatchify components.
+`HatchifyProvider` is a component that hosts and provides access to Hatchify-related state. It must be a parent to any Hatchify components.
 
 ```tsx
-const App: React.FC = () => {
+import {HatchifyProvider} from "@hatchifyjs/react"
+
+// Define Schemas
+
+const schemas = { ... }
+
+// Create the Hatched React App instance
+const hatchedReact = hatchifyReact(createJsonapiClient("/api", Schemas))
+
+// Define variables for your Hatchify schema generated components
+const TodoDataGrid = hatchedReact.components.Todo.DataGrid
+
+const HatchedComponent = () => {
   return (
-    <ThemeProvider theme={createTheme()}>
-      <HatchifyProvider>
-        {/* 👀 */}
-        <Everything />
-      </HatchifyProvider>
-    </ThemeProvider>
+    <HatchifyProvider>
+      <TodoDataGrid defaultSort={{ direction: "asc" }} />
+    </HatchifyProvider>
   )
 }
 ```
 
-## hatchifyReact
+### Types
+
+`@hatchifyjs/react` provides a number of [types](./types.md) to assist with app customization.
 
-`hatchifyReact` is a function that initializes the `HatchifyApp` object. Inside of the object you will find `components`, `model`, `state`, and `Everything`. `components`, `model`, and `state`, will be broken down further for each defined schema
+| key                                         | description                                                                             |
+| ------------------------------------------- | --------------------------------------------------------------------------------------- |
+| [`DataGridState`](./types.md#datagridstate) | The return type of the `useDataGridState` hook.                                         |
+| [`CreateType`](./types.md#createtype)       | The type used when data is created.                                                     |
+| [`UpdateType`](./types.md#updatetype)       | Is the type used when data is updated.                                                  |
+| [`HatchifyApp`](./types.md#hatchifyapp)     | the type of the Hatchify App that is generated with [`hatchifyReact()`](#hatchifyreact) |
+| [`RecordType`](./types.md#recordtype)       | Matches a record's schema                                                               |
```

Comment:
for the details bit i'd say:

DataGridState - The return type of `DataGridState`. Can be used if you are extending the functionality of `useDataGridState`.

CreateType - This can be used to type data that will be used to create a record for a given schema

UpdateType - ^ same as above but "update" instead of "create"

HatchifyApp - The return type of the object that is generated by `hatchifyReact`

RecordType - The type of a record for a given schema